controllers/openstack_ansibleee_controller.go

```diff
@@ -208,22 +208,30 @@ func (r *OpenStackAnsibleEEReconciler) Reconcile(ctx context.Context, req ctrl.R
 
 	if (ctrlResult != ctrl.Result{}) {
 		instance.Status.Conditions.Set(condition.FalseCondition(
-			ansibleeev1.AnsibleExecutionJobReadyCondition,
+			condition.JobReadyCondition,
 			condition.RequestedReason,
 			condition.SeverityInfo,
-			ansibleeev1.AnsibleExecutionJobWaitingMessage))
+			condition.JobReadyRunningMessage))
 		instance.Status.JobStatus = ansibleeev1.JobStatusRunning
 		return ctrlResult, nil
 	}
 
 	if err != nil {
+		var errorReason condition.Reason
+		errorReason = condition.ErrorReason
+		severity := condition.SeverityWarning
+		if ansibleeeJob.HasReachedLimit() {
+			errorReason = condition.JobReasonBackoffLimitExceeded
+			severity = condition.SeverityError
+		}
 		instance.Status.Conditions.Set(condition.FalseCondition(
-			ansibleeev1.AnsibleExecutionJobReadyCondition,
-			condition.ErrorReason,
-			condition.SeverityWarning,
-			ansibleeev1.AnsibleExecutionJobErrorMessage,
+			condition.JobReadyCondition,
+			errorReason,
+			severity,
+			condition.JobReadyErrorMessage,
 			err.Error()))
 		instance.Status.JobStatus = ansibleeev1.JobStatusFailed
+		instance.Status.Failed = ansibleeeJob.GetTotalFailedAttempts()
```

Comment:
What scenarios pods won't be retried, so we get the value here different from backoffLimit we set the OpenStackDataPlaneDeployment(job) spec (default 6). Just wondering if there is value in keeping track of this here unless we allow to changing `podFailurePolicy` for jobs?

Reply:
this is for tracking each attempt:
attempt 1 failed
attempt 2 failed
...
the idea is to get more transparency on the retries: [OSPRH-6593](https://issues.redhat.com/browse/OSPRH-6593)
https://github.com/openstack-k8s-operators/dataplane-operator/pull/879/files#r1611441361

Reply:
Why would we want to add every retry failure to the status unless it has some details that's different from every retry?

Reply:
I think exposing the retry number shows some progress

Reply:
I understand the benefit, but I'm unsure if status is the right place to put it.

In principle it makes sense to let people know how many attempts have been made. No doubt about that. But my understanding of `status` is that it isn't concerned that much about history, as it is about ... well *state*. 

Personally I wouldn't agonize over it. At most we are talking about style and practice, not about functionality.

Reply:
I've dropped the `failed` field